### PR TITLE
Space Travelers' Hub

### PR DESCRIPTION
**Pull request by mhnazary**
# 🏅 Features Implemented
_Project Requirements -> https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md_

In this project, we worked with real live data from the SpaceX API. We built a web application for a company that provides commercial and scientific space travel services. The application allows users to book rockets and join selected space missions.

List of things that were implemented based on the requirements:

## 📝 General requirements

- ✅ No Linters errors.
- ✅ Used GitFlow correctly.
- ✅ Created a professional `README` file.

## 📝 HTML/CSS requirements

- ✅ Followed the best HTML/CSS practices.

## 📝 JavaScript requirements

- ✅ Followed the best JavaScript practices.

## 📝 Project requirements

We created PR for every step like this:

- ✅ Config & basic setup.
- ✅ Redux: Fetch data and update the Redux store.
- ✅ Render UI: lists.
- ✅ Redux: Write actions and reducers for canceling rockets/dragons and leaving missions.
- ✅ Render UI: conditional components rendering.
- ✅ Render UI: My Profile section.
- ✅ Added test Using jest library.

## 📷 Screenshots

image
image
image

_John Palacios & Mahram Hossain_
_Microverse Student_

### CODE REVIEW

File: src/test/Missions/Mission.test.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import '@testing-library/jest-dom/extend-expect';
```
Comment (gemmen29) [inline src/test/Missions/Mission.test.jsx:1]: - [ ] Good job on adding tests. Kindly make sure to add tests for all your components and functions as per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/group-project/project_space_travelers_hub.md#:~:text=NOTE%3A%20Make%20sure%20to%20add%20test%20for%20all%20your%20components%20and%20functions.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/test/Missions/Mission.test.jsx (hunk 1 above)
Comment (Rysth) [reply src/test/Missions/Mission.test.jsx]: @gemmen29 Thanks!! Done.

**Review by muneebulrehman — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- All tests passing
- Good design
- All functionality working as expected
- Good follow-up with the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**